File: backend/features/deployment/models/git_hub_project.js
const mongoose = require('mongoose');

const gitHubProjectSchema = mongoose.Schema({
    fullName: {
        type: String,
        required: true
    },
    cloneUrl: {
        type: String,
        required: true
    },
    ownerName: {
        type: String,
        required: true
    },
    repoName: {
        type: String,
        required: true
    },
    branchName: {
        type: String,
        required: true
    }
}, {
    timestamps: true
});

gitHubProjectSchema.set('toJSON', {
  transform: function (doc, ret) {
    if (ret.createdAt) {
      ret.createdAt = ret.createdAt.toISOString();
    }
    if (ret.updatedAt) {
      ret.updatedAt = ret.updatedAt.toISOString();
    }
    return ret;
  }
});

const GitHubProject = mongoose.model('GitHubProject', gitHubProjectSchema);
module.exports = GitHubProject;

File: frontend/lib/features/projects/domain/entities/git_hub_project.dart
class GitHubProject {
  final String fullName;
  final String cloneUrl;
  final String ownerName;
  final String repoName;
  final String branchName;
  final DateTime createdAt;

  GitHubProject({
    required this.fullName,
    required this.cloneUrl,
    required this.ownerName,
    required this.repoName,
    required this.branchName,
    required this.createdAt,
  });

  @override
  String toString() {
    return 'GitHubProject{fullName: $fullName, cloneUrl: $cloneUrl, ownerName: $ownerName, repoName: $repoName, branchName: $branchName, createdAt: ${createdAt.toIso8601String()}}';
  }
}
